lib/matplotlib/widgets.py

```diff
@@ -973,45 +1048,259 @@ def clear(self, event):
+
+        for prop, new_prop in zip(properties, self.properties):
+            new_prop['visible'] = True
+            new_prop['horizOn'] = prop.pop('horizOn', True)
+            new_prop['vertOn'] = prop.pop('vertOn', True)
+
+            new_prop['coord'] = np.array([
+                0.5*(ax.get_xbound()[0] + ax.get_xbound()[1]),
+                0.5*(ax.get_ybound()[0] + ax.get_ybound()[1])
+                ])
+
+            if self.useblit:
+                prop['animated'] = True
+
+            new_prop['lineh'] = ax.axhline(
+                new_prop['coord'][1],
+                visible=False,
+                **prop)
+            new_prop['linev'] = ax.axvline(
+                new_prop['coord'][0],
+                visible=False,
+                **prop)
+
+        self.__selected = None
+        self._draw_cursor()
 
-        self._update()
 
     def _update(self):
 
         if self.useblit:
             if self.background is not None:
                 self.canvas.restore_region(self.background)
-            self.ax.draw_artist(self.linev)
-            self.ax.draw_artist(self.lineh)
+            for prop in self.properties:
+                self.ax.draw_artist(prop['linev'])
+                self.ax.draw_artist(prop['lineh'])
             self.canvas.blit(self.ax.bbox)
         else:
-
             self.canvas.draw_idle()
 
         return False
 
+    def _draw_cursor(self):
+        """draw cursor specified by dictionary entries in prop"""
+
+        self.needclear = True
+
+        for prop in self.properties:
+
+            if prop['visible']:
+
+                prop['linev'].set_xdata((prop['coord'][0], prop['coord'][0]))
+                prop['lineh'].set_ydata((prop['coord'][1], prop['coord'][1]))
+
+                prop['linev'].set_visible(prop['vertOn'])
+                prop['lineh'].set_visible(prop['horizOn'])
+
+        self._update()
+
+    def coord(self, cursor):
+        """
+        current coordinates of the cursor
+        """
+        return self.properties[cursor]['coord'].copy()
+
+    def clear(self, event):
+        """clear the cursor"""
+        if self.ignore(event):
+            return
+        if self.useblit:
+            self.background = self.canvas.copy_from_bbox(self.ax.bbox)
+        for prop in self.properties:
+            prop['linev'].set_visible(False)
+            prop['lineh'].set_visible(False)
+
+    def on_press(self, event):
+        """
+        on left click we will see if the mouse is over us
+        and store some data
+        """
+
+        # Verify left click
+        if event.button != 1:
+            return
+
+        # Verify in axis
+        if event.inaxes != self.ax:
+            return
+
+        # Find cursor closest to click location
+        # the best user behavior is to look at the closest visible line
+        min_distance = -1.0
+        for prop in self.properties:
+            if prop['visible']:
+                if prop['vertOn']:
```

Comment:
it is not immediately obvious to me that this is better than computing the actual euclidean distance from the center.  This selects the cursor that the mouse is closest to in orthogonal distance to any visible part where as it seems more 'natural' (which I acknowledge is very subjective) to go to the cursor that the the mouse is closest to the center of.  On the other hand, this is way I do not think we need a notion of 'maximum' distance away from the cursor at which the click is ignored and this way obviously works if only one of the lines is present.

Which is mostly me talking myself into as implemented being the correct way to do this which I will leave here in 
case anyone else has the same concerns.